File: abc/abc209/b.py
# https://atcoder.jp/contests/abc209/submissions/37372030

n, x = map(int, input().split())
a = list(map(int, input().split()))

s = 0
for i in range(n):
  if i % 2:
    a[i] -= 1
  s += a[i]

print("No" if s > x else "Yes")

File: abc/abc209/b.cpp
// https://atcoder.jp/contests/abc209/submissions/26985363

#include <bits/stdc++.h>
using namespace std;
int main() {
  int n, x, s = 0, count = 1;  cin >> n >> x;
  while(cin >> n) {if(count % 2 == 0) s += (n - 1); else s += n; count++;}
  if(s > x) cout << "No";
  else cout << "Yes";
}
